src/App.tsx

```diff
@@ -22,30 +22,61 @@ const Header = (
 );
 
 const App: React.FC = () => {
-  const [victimList, setVictimList] = useState([]);
+  const [victimList, setVictimList] = useState();
   const [isDataReady, setIsDataReady] = useState(true);
-  const [isResponseOk, setIsResponseOk] = useState(true);
+  const [isResponseOk, setIsResponseOk] = useState();
 
   const fetchDetails = async (data: SearchData) => {
     setIsDataReady(false);
     console.log(
       "sending request to: ",
       process.env.REACT_APP_BACKEND_URL + `/find/victim/byName/${data.name}`
     );
-    const response = await fetch(
+    fetch(
       process.env.REACT_APP_BACKEND_URL + `/find/victim/byName/${data.name}`
-    );
+    )
+    .then((response) => {
+      if (response.ok) {
+        // Returning the JSON response if HTTP status response code is 200/201
+        const contentType = response.headers.get("content-type");
+        if (contentType && contentType.indexOf("application/json") !== -1) {
+          // Returns JSON object if the response is of type JSON.
+          return response.json();
+        } else {
+          // Returns string if the response is of type string.
+          return response.text();
+        }
+      } else {
+        // Catch other HTTP status response.
+          setVictimList(null);
```

Comment:
This is the only solution for the No data display error message.

Since, It returns an empty array onload of the page and also the condition in our typescript checks for the length. We faced that No data display error.

We can alter it in future refactoring with different approach.